src/Microsoft.AspNetCore.Routing/Matchers/AsciiKeyedJumpTable.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    // An optimized jump table that trades a small amount of additional memory for
+    // hash-table like performance.
+    //
+    // The optimization here is to use the first character of the known entries
+    // as a 'key' in the hash table in the space of A-Z. This gives us a maximum
+    // of 26 buckets (hence the reduced memory)
+    internal class AsciiKeyedJumpTable : JumpTable
+    {
+        public static bool TryCreate(
+            int @default,
```

Comment:
These are indexes being passed in? Worth suffixing with Index?

Reply:
This is being cleaned up as part of #597. I'll remove that class from this PR.